tests/e2e/plugins/index.js

```diff
@@ -5,20 +5,48 @@
 // as explained in the cypress docs
 // https://docs.cypress.io/api/plugins/preprocessors-api.html#Examples
 
-/* eslint-disable global-require, arrow-body-style */
-// const webpack = require('@cypress/webpack-preprocessor')
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const webpack = require('@cypress/webpack-preprocessor');
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const vueCliOptions = require('@vue/cli-service/webpack.config');
+
+/**
+ * We must use a different webpack congiuration for cypress. This is because
+ * we need to use a different tsconfig.json file to compile the .ts files.
```

Comment:
> we need to use a different tsconfig.json file

why?

Reply:
We need to add `cypress` to the `types`. Otherwise, it does not understand what `cy` or `Cypress` are. Also, I can't add it to the root `tsconfig.json` because it collides with `jest`. They both share `describe`, `it`, etc.

Reply:
ah yup. that checks with our tsconfig at work.